libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/config/SyncsConfig.kt

```diff
@@ -0,0 +1,158 @@
+     * @param ctx Context.
+     * @param resourceId Id of resource file.
+     */
+    constructor(ctx: Context?, resourceId: Int) : this(
+        ResourceReaderHelper.readResourceFile(
+            ctx,
+            resourceId
+        )
+    )
+
+    /**
+     * Constructor for config stored in asset file
+     * @param ctx Context.
+     * @param assetPath Path of assets file.
+     */
+    constructor(ctx: Context?, assetPath: String?) : this(
+        ResourceReaderHelper.readAssetFile(
+            ctx,
+            assetPath
+        )
+    )
+
+    init {
+        try {
+            syncConfigs = if (str == null) {
+                null
+            } else {
+                val config = JSONObject(str)
+                config.getJSONArray(SYNCS)
+            }
+        } catch (e: JSONException) {
+            MobileSyncLogger.e(TAG, "Unhandled exception parsing json", e)
+        }
+    }
+
+    /**
+     * Return true if syncs are defined in config
+     * @return
+     */
+    fun hasSyncs(): Boolean {
+        return syncConfigs?.let { it.length() > 0 } == true
+    }
+
+    /**
+     * Create the syncs from the config in the given store
+     * NB: only feedback is through the logs - the config is static so getting it right is something the developer should do while writing the app
+     * @param store
+     */
+    fun createSyncs(store: SmartStore) {
+        val syncConfigs = syncConfigs ?: return
+        val syncManager: SyncManager = SyncManager.getInstance(null, null, store)
+        JSONObjectHelper.toList<JSONObject>(syncConfigs).forEach { syncConfig ->
+            try {
+                val syncName = syncConfig.getString(SYNC_NAME)
+                // Leaving sync alone if it already exists
+                if (syncManager.hasSyncWithName(syncName)) {
+                    MobileSyncLogger.d(TAG, "Sync already exists:$syncName - skipping")
+                }
+                else {
+                    val syncType = SyncState.Type.valueOf(syncConfig.getString(SYNC_TYPE))
+                    val targetJson = syncConfig.optJSONObject(TARGET)
+                        ?: throw SyncManager.MobileSyncException("Target not defined in config")
+                    val optionsJson = syncConfig.optJSONObject(OPTIONS)
+                        ?: throw SyncManager.MobileSyncException("Options not defined in config")
+                    val options: SyncOptions = SyncOptions.fromJSON(optionsJson)
+                    val soupName = syncConfig.getString(StoreConfig.SOUP_NAME)
+                    MobileSyncLogger.d(TAG, "Creating sync:$syncName")
+                    when (syncType) {
+                        SyncState.Type.syncDown -> {
+                            val target = SyncDownTarget.fromJSON(targetJson)
+                            syncManager.createSyncDown(target, options, soupName, syncName)
+                        }
+
+                        SyncState.Type.syncUp -> {
```

Comment:
Optional: Static imports for `syncUp` and `syncDown` could aid readability.  Are they part of our public API or could they be renamed with the conventional leading uppercase?